core/src/mindustry/ai/types/PayloadAI.java

```diff
@@ -0,0 +1,48 @@
+package mindustry.ai.types;
+
+import arc.math.geom.*;
+import mindustry.entities.*;
+import mindustry.gen.*;
+import mindustry.world.blocks.*;
+
+public class PayloadAI extends FlyingAI{
```

Comment:
Haven't merged this yet because I really don't like the idea of there being a special AI for payloads.

Reply:
> Haven't merged this yet because I really don't like the idea of there being a special AI for payloads.

payload units are special, it would make sense for them to have special ai. unless your meaning specifically units carrying payloads. which then I would just have them drop their payload upon meeting some conditions (about to die, near enemy defenses, something else idk).

Reply:
As long as i got it correctly, you especifically asked how would it work in AI terms, so i got to work in an AI. 
[this message actually](https://discord.com/channels/391020510269669376/663853133587546138/765928704727449650)

Reply:
If you want me to remove the AI, i can perfectly do so, completely up to you